$OpenBSD: patch-libcli_auth_schannel_state_tdb_c,v 1.1 2015/02/25 01:37:49 jca Exp $

- CVE-2015-0240: talloc free on uninitialized stack pointer in netlogon
  server could lead to remote-code execution
  https://bugzilla.samba.org/show_bug.cgi?id=11077#c32

--- libcli/auth/schannel_state_tdb.c.orig	Wed May  8 10:16:26 2013
+++ libcli/auth/schannel_state_tdb.c	Mon Feb 23 19:52:53 2015
@@ -285,6 +285,10 @@ NTSTATUS schannel_check_creds_state(TALLOC_CTX *mem_ct
 	NTSTATUS status;
 	int ret;
 
+	if (creds_out != NULL) {
+		*creds_out = NULL;
+	}
+
 	tmpctx = talloc_named(mem_ctx, 0, "schannel_check_creds_state");
 	if (!tmpctx) {
 		return NT_STATUS_NO_MEMORY;
